pkg/datapath/xdp/xdp.go

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package xdp
+
+import (
+	"fmt"
+
+	"github.com/cilium/ebpf/link"
+	"github.com/cilium/hive/cell"
+)
+
+// Mode represents the mode to use when loading XDP programs.
+// They are wrappers around the underlying mode names passed to
+// the traffic control (tc) system in the kernel.
+type Mode string
+
+const (
+	// ModeNative for loading progs with TCModeLinkDriver
+	ModeNative Mode = "native"
+
+	// ModeBestEffort for loading progs with TCModeLinkDriver,
+	// falling back to TCModeLinkGeneric if unsupported by the
+	// underlying device.
+	ModeBestEffort Mode = "best-effort"
+
+	// ModeGeneric for loading progs with TCModeLinkGeneric
+	ModeGeneric Mode = "testing-only"
+
+	// ModeDisabled for not having XDP enabled
+	ModeDisabled Mode = "disabled"
+)
+
+// TCMode represents the name of an XDP mode from the perspective
+// of the traffic control (tc) system in the kernel.
+type TCMode string
```

Comment:
Technically TC and XDP have nothing to do with each other so the way its explained is a bit confusing. I would personally rename `Mode` to `AccelerationMode` and `TCMode` to `Mode`. Since `native`/`best-effort`/`disabled` is talking about accelerating part of the datapath code. And `xdpdrv`/`xdpgeneric` is the mode in which the XDP program is loaded.

Reply:
That's a good idea. I wasn't too happy with `TCMode` but was struggling to think of an alternative. Your suggestion is perfect :ok_hand: